src/com/facebook/buck/swift/SwiftRuntimeNativeLinkable.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.swift;
+
+import static com.facebook.buck.swift.SwiftDescriptions.isSharedRequested;
+
+import com.facebook.buck.apple.AppleCxxPlatform;
+import com.facebook.buck.cxx.CxxPlatform;
+import com.facebook.buck.cxx.Linker;
+import com.facebook.buck.cxx.NativeLinkable;
+import com.facebook.buck.cxx.NativeLinkableInput;
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.model.Flavor;
+import com.facebook.buck.model.ImmutableFlavor;
+import com.facebook.buck.parser.NoSuchBuildTargetException;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.args.StringArg;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.nio.file.Path;
+
+/**
+ * Pseudo linkable for representing Swift runtime library's linker arguments.
+ */
+final class SwiftRuntimeNativeLinkable implements NativeLinkable {
```

Comment:
The pseudo build target is a problem.

How was using the native linkable solving the duplicate flags before they were all using the same pseudo build target?

Would adding the flags at the binary level work? Although that's going to introduce a massive dependency cycle cxx -> swift -> apple -> cxx.

Alternatively can we create a type of NativeLinkable that doesn't have a build target? But then how do you identify it to deduplicate it?

Still thinking about this.